src/main/webapp/script.js

```diff
@@ -47,22 +55,73 @@ function initializeRow(json, key, currentRow) {
   let index = 0;
   for (const type of json[key]) {
     const columnEntry = document.createElement('td');
```

Comment:
Do you need to check whether the json data has this key?
Also rename the parameter names to make it more meaningful.

Reply:
this comment is might be best answered by @D-moe  since its his code, but i can go ahead and give the variables/functions better names

Reply:
Eslint requires a check to determine if the json data has a key. I was surprised by this too but it seems that it is possible to get inside the loop without this condition being true.